File: da-test/src/test/java/ro/johann/da/test/api/service/decision/transfer/SetPropertyInput.java
package ro.johann.da.test.api.service.decision.transfer;

import lombok.Builder;
import lombok.Value;
import lombok.experimental.FieldDefaults;

import static lombok.AccessLevel.PRIVATE;

@Value
@Builder(builderClassName = "Builder")
@FieldDefaults(level = PRIVATE)
public class SetPropertyInput {
  String alternativeId;
  String criteriaId;
  Float value;
}

File: da-decision-service/src/main/kotlin/ro/johann/da/decision/api/transfer/SetPropertyInput.kt
package ro.johann.da.decision.api.transfer

import java.util.UUID
import javax.validation.constraints.NotNull

data class SetPropertyInput(

  @field:NotNull
  val alternativeId: UUID,

  @field:NotNull
  val criteriaId: UUID,

  @field:NotNull
  val value: Double
)
